src/features/posts.ts

```diff
@@ -0,0 +1,53 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostState = {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const loadPosts = createAsyncThunk(
+  'posts/fetch',
+  async (userId: number) => {
+    const result = await getUserPosts(userId);
+
+    return result;
+  },
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearPosts: state => {
+      return { ...state, posts: [] };
```

Comment:
In the `clearPosts` reducer, you are returning a new state object with a `posts` property. However, your state object doesn't have a `posts` property, it has an `items` property. So, this code won't clear your posts as expected. You should return `{ ...state, items: [] }` instead.